dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -496,3 +507,96 @@ fn load_manifest(path: &Path) -> Result<DscResource, DscError> {
 
     Ok(resource)
 }
+
+fn sort_adapters_based_on_lookup_table(unsorted_adapters: &BTreeMap<String, Vec<DscResource>>, needed_resource_types: &Vec<String>) -> LinkedHashMap<String, Vec<DscResource>>
+{
+    let mut result:LinkedHashMap<String, Vec<DscResource>> = LinkedHashMap::new();
+    let lookup_table:HashMap<String, String> = load_adapted_resources_lookup_table();
+    // first add adapters (for needed types) that can be found in the lookup table
+    for needed_resource in needed_resource_types {
+        if let Some(adapter_name) = lookup_table.get(needed_resource) {
+            if let Some(resource_vec) = unsorted_adapters.get(adapter_name) {
+                trace!("Lookup table found resource '{}' in adapter '{}'", needed_resource, adapter_name);
+                result.insert(adapter_name.to_string(), resource_vec.clone());
+            }
+        }
+    }
+
+    // now add remaining adapters
+    for (adapter_name, adapters) in unsorted_adapters {
+        if !result.contains_key(adapter_name) {
+            result.insert(adapter_name.to_string(), adapters.clone());
+        }
+    }
+
+    result
+}
+
+fn add_resources_to_lookup_table(adapted_resources: &BTreeMap<String, Vec<DscResource>>)
+{
+    let mut lookup_table:HashMap<String, String> = load_adapted_resources_lookup_table();
+
+    for (resource_name, res_vec) in adapted_resources {
+        let adapter_name = &res_vec[0].require_adapter.as_ref().unwrap();
+        lookup_table.insert(resource_name.to_string().to_lowercase(), (*adapter_name).to_string());
+    };
+
+    save_adapted_resources_lookup_table(&lookup_table);
+}
+
+fn save_adapted_resources_lookup_table(lookup_table: &HashMap<String, String>)
+{
+    if let Ok(lookup_table_json) = serde_json::to_string_pretty(&lookup_table) {
```

Comment:
We should not save this as pretty since it probably has a minor perf impact to write and read.  When debugging, we can use other tools to pretty print it.

Reply:
Change `to_string_pretty` -> `to_string`.